contents/testing/writing-testable-code.md

```diff
@@ -0,0 +1,146 @@
+# Writing Testable Code
+
+When the project is big enough and needs to be maintainable in the long run, it has to rely on automated tests to keep its quality. Especially, the idea of unit testing becomes very important for Object-Oriented programs. Compared to system testing where you test the program altogether, unit testing has its benefit for being fast (because it only instantiates a small piece of the program) and stable (because it usually mocks out the unstable dependency, e.g. network connection, database connection).
```

Comment:
> Don't we rely on automated tests to prevent regression, not keep quality?

Having automation helps to prevent regression. As a result, we also keep up the code quality, right?





> Especially is not the right word to use here. Maybe you can use "Because of this,".

Thanks. Decide to use "Because of this", but moving the sentence to the end.